### DISCUSSION TITLE
Information to help make GFX Library ESP32 Boards 3.0.x compatible

### DISCUSSION THREAD

**Discussion by jfseaman** [moononournation/Arduino_GFX#538] [Ideas]
I am willing to help with coding....

I acquired a ESP32-S3-LCD-Driver board (Waveshare/Spotbear) https://www.waveshare.com/esp32-s3-lcd-driver-board.htm

This seems to be a "new" board as all the demo programs/sketches require latest Arduino-IDE board definitions/ESP-IDE SDK. As of 2004/9/1, those are 3.0.5 and 5.3 respectively .

It took a bit of head scratching but I got the Arduino-IDE and VSCode+Espressif extension examples working.  https://files.waveshare.com/wiki/ESP32-S3-LCD-Driver-Board/ESP32-S3-LCD-Driver-Board/Demo/ESP32-S3-LCD-Driver-Board_demo.zip

These examples use the "new" Espressif libraries: 
ESP_Display_Panel https://github.com/esp-arduino-libs/ESP32_Display_Panel
ESP_IO_Expander https://github.com/esp-arduino-libs/ESP32_IO_Expander

The libraries use the new esp_memory_utils.h found at {Arduino-IDE location}\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.1-33fbade6\esp32s3\include\esp_hw_support\include and {espressif location]/esp-idf-v5.3\components\esp_hw_support\include

Perhaps the keys to making GFX_Library support ESP32 RGB displays can be found in these libraries and headers. 

More useful information can be found at https://www.waveshare.com/wiki/ESP32-S3-LCD-Driver-Board#Arduino_Sample_Demos

**Comment by mrmonteith** [comment]
I just bought an8048S070C 7" display.  The problem is hard to manage multiple projects on the IDE when libraries have to have varied versions.  I've had the display too long to return and spent too many hours trying to get it going.   Probably go in the box on the shelf for stuff maybe later.
